src/operation/in_memory_model_store.jl

```diff
@@ -1,74 +1,12 @@
-# So far just the EmulationStore is implemented.
-
-"""
-Stores results data for one DecisionModel
-"""
-mutable struct DecisionModelOptimizerResults <: AbstractModelOptimizerResults
-    duals::Dict{ConstraintKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}}
-    parameters::Dict{ParameterKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}}
-    variables::Dict{VariableKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}}
-    aux_variables::Dict{AuxVarKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}}
-    expressions::Dict{ExpressionKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}}
-end
-
-function DecisionModelOptimizerResults()
-    return DecisionModelOptimizerResults(
-        Dict{
-            ConstraintKey,
-            Dict{ConstraintKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}},
-        }(),
-        Dict{
-            ParameterKey,
-            Dict{ParameterKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}},
-        }(),
-        Dict{
-            VariableKey,
-            Dict{VariableKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}},
-        }(),
-        Dict{
-            AuxVarKey,
-            Dict{ConstraintKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}},
-        }(),
-        Dict{
-            AuxVarKey,
-            Dict{ExpressionKey, OrderedDict{Dates.DateTime, DataFrames.DataFrame}},
-        }(),
-    )
-end
-
-"""
-Stores results data for one EmulationModel
-"""
-mutable struct EmulationModelOptimizerResults <: AbstractModelOptimizerResults
-    last_recorded_row::Int
-    duals::Dict{ConstraintKey, DataFrames.DataFrame}
-    parameters::Dict{ParameterKey, DataFrames.DataFrame}
-    variables::Dict{VariableKey, DataFrames.DataFrame}
-    aux_variables::Dict{AuxVarKey, DataFrames.DataFrame}
-    expressions::Dict{ExpressionKey, DataFrames.DataFrame}
-end
-
-function EmulationModelOptimizerResults()
-    return EmulationModelOptimizerResults(
-        0,
-        Dict{ConstraintKey, DataFrames.DataFrame}(),
-        Dict{ParameterKey, DataFrames.DataFrame}(),
-        Dict{VariableKey, DataFrames.DataFrame}(),
-        Dict{AuxVarKey, DataFrames.DataFrame}(),
-        Dict{ExpressionKey, DataFrames.DataFrame}(),
-    )
-end
-
 """
 Stores simulation data in memory
 """
 mutable struct InMemoryModelStore{T <: AbstractModelOptimizerResults}
     data::T
-    optimizer_stats::OrderedDict{Int, OptimizerStats}
 end
 
 function InMemoryModelStore(::Type{T}) where {T <: AbstractModelOptimizerResults}
-    return InMemoryModelStore(T(), OrderedDict{Int, OptimizerStats}())
+    return InMemoryModelStore(T())
```

Comment:
This file is now almost useless. If we moved the initialize function to each of the subtypes then each model can simply create the desired type.

Reply:
we can do that if you prefer. Probably the design was over complicated.